### PR TITLE
output gas pressure for gas-water case

### PR DESCRIPTION
**Pull request by totto82**
Issue reported in #4316

### CODE REVIEW

**Comment by totto82** [comment]
jenkins build this please

File: ebos/ecloutputblackoilmodule.hh [hunk 1]
```diff
@@ -234,11 +234,11 @@ public:
...
-                    // put pressure in oil pressure for output
-                    if (FluidSystem::phaseIsActive(waterPhaseIdx)) {
-                        this->oilPressure_[globalDofIdx] = getValue(fs.pressure(waterPhaseIdx));
-                    } else {
+                    // put pressure in oil pressure for output. Use gas if oil is not present
+                    if (FluidSystem::phaseIsActive(gasPhaseIdx)) {
                         this->oilPressure_[globalDofIdx] = getValue(fs.pressure(gasPhaseIdx));
+                    } else { // use water if neither oil nor gas is present
+                        this->oilPressure_[globalDofIdx] = getValue(fs.pressure(waterPhaseIdx));
                     }
```
Comment (bska) [inline ebos/ecloutputblackoilmodule.hh:242]: If we're changing this anyway, I think I'd prefer that we straighten out the conditionals too, e.g.,
```C++
if (FluidSystem::phaseIsActive(oilPhaseIdx)) {
    // ...
}
else if (FluidSystem::phaseIsActive(gasPhaseIdx)) {
    // ...
}
else {
    // ...
}
```
We might even consider writing this as a loop
```C++
for (const auto idx : { oilPhaseIdx, gasPhaseIdx, waterPhaseIdx }) {
    if (FluidSystem::phaseIsActive(idx)) {
        this->oilPressure_[globalDofIdx] = getValue(fs.pressure(idx));
        break;
    }
}
```
but that might on the other hand be too obscure.

**Review by bska — COMMENTED** [review]
Looks good to me.  I just think we should take the opportunity to reduce the level of nesting here.   I'll merge once that's done.

File: ebos/ecloutputblackoilmodule.hh (hunk 1 above)
Comment (totto82) [reply ebos/ecloutputblackoilmodule.hh:242]: I think we should also rename oilPressure_. What about referencePressure_?

Comment (bska) [reply ebos/ecloutputblackoilmodule.hh:242]: > I think we should also rename `oilPressure_`

I think so too.  Explicitly referring to this as oil pressure is likely to be more confusing as time passes.

> What about `referencePressure_`?

Yeah, either that or maybe `fluidPressure_`.

Comment (totto82) [reply ebos/ecloutputblackoilmodule.hh]: Good suggestion.

**Comment by totto82** [comment]
jenkins build this please

**Review by bska — APPROVED** [review]
Very good.  I'll merge into master.

**Pull request merged**